WcaOnRails/app/controllers/registrations_controller.rb

```diff
@@ -81,12 +81,21 @@ def edit
   end
 
   def destroy
-    @registration = Registration.find(params[:id])
-    @registration.destroy
-    mailer = RegistrationsMailer.notify_registrant_of_deleted_registration(@registration)
-    mailer.deliver_now
-    flash[:success] = "Deleted registration and emailed #{mailer.to.join(" ")}"
-    redirect_to competition_edit_registrations_path(@registration.competition)
+    @competition = competition_from_params
+    if params.has_key?(:user_is_deleting_theirself)
+      registrations = @competition.registrations
+      @registration = registrations.find_by_user_id(current_user.id)
```

Comment:
For consistency's sake (we have an email for almost every other registration action, I believe), I believe we should have an email for this. It would probably make sense to email the organizers and let them know what happened.